teps/0129-multiple-tekton-instances-per-cluster.md

```diff
@@ -95,40 +95,168 @@ bundled in the Tekton webhook as well the remote resolution controller.
 
 ## Proposal
 
-<!--
-This is where we get down to the specifics of what the proposal actually is.
-This should have enough detail that reviewers can understand exactly what
-you're proposing, but should not include things like API designs or
-implementation. The "Design Details" section below is for the real
-nitty-gritty.
--->
+Tekton is design as an extension of Kubernetes through the [Custom Resource Definition][crd]
+mechanism and, as such, it must abide by the constraints associated with it.
+The most relevant constraint in terms of this TEP is that custom resources (CRDs)
+are defined at the cluster level. As such, CRDs and components that operate on them can only
+be installed once in a Kubernetes cluster, and must be shared across multiple installations
+of Tekton within the same cluster.
+
+In this TEP we propose to separate such components so that it becomes possible for a Tekton
+installer system to install, update and uninstall the parts of Tekton required for multiple
+installation of Tekton to coexist within a single cluster.
+
+The components in questions are:
+- The CRDs themselves. These are defined in dedicated files today, so no action needed for them
+- The conversion controller: this controller is responsible for converting the API version
+  of resources on the fly when they read and written from and to the cluster, whenever the
+  version used by the client is served but not the one stored in etcd. The conversion
+  controller is based on [`knative/pkg`][knative-convertion] implementation, and it's also
+  responsible for [reconciling and altering the CRD definition][crd-reconcile].
+  Today in Tekton Pipelines this controller is [bundled together][controller-bundle] with
+  other admission controllers in a single binary. 
+- Service accounts, roles and bindings (RBAC) required by the conversion controller to 
+  operate - specifically read/write access to the CRDs.
```

Comment:
Those 3 components are the ones that are considered "singleton", am I right ?
One question, why separating the conversion controller from the other admission controllers ? They *all* need to be "singleton" as well, don't they ?

Reply:
The conversion controller needs read/write access to the CRDs, which are cluster-level.
The admission controllers can be configured to be namespace specific, so that each instance of Tekton may retain its own webhooks, thus reducing the risk of issues.

Reply:
Ah understood 👼🏼 I think it would make sense to highlight this in the proposal 👼🏼

Reply:
Done